android/src/main/java/io/fullstack/firestack/database/FirestackDatabaseReference.java

```diff
@@ -114,105 +114,73 @@ public void onCancelled(DatabaseError error) {
         err.putInt("errorCode", error.getCode());
         err.putString("errorDetails", error.getDetails());
         err.putString("description", error.getMessage());
+        mOnceValueListeners.remove(this);
         callback.invoke(err);
       }
     };
 
-    Query ref = this.getDatabaseQueryAtPathAndModifiers(modifiersArray);
-    ref.addListenerForSingleValueEvent(mOnceValueListener);
+    //TODO ? Is it really necessary to track this type of event listener as they are automatically
```

Comment:
@chrisbianca you're right - I think this is pointless really, as with any subscriber/events module 'once' will instantly unsubscribe on first event - and as firebase always sends the initial data as a first event this is guaranteed to be fired.

Feel free to remove as the event emitter on the JS side can also handle it's own once logic, so deffo not needed.